lib/message.js

```diff
@@ -39,12 +54,112 @@ var Message = function (client) {
 		.then(function (response) {
 			var message = params;
 			var location = response.headers.location;
-			var messageId = location.substring(location.lastIndexOf("/") + 1);
-			message.id = messageId;
+			message.id = location.substring(location.lastIndexOf("/") + 1);
 			return message;
 		})
 		.asCallback(callback);
 	};
+	/**
+	 * Send multiple SMS or MMS messages with one API call.
+	 * This is much more performant than calling `send` multiple times.
+	 * @param  {Array} params An array of params objects, each of which
+	 * represents a single text message. The returned array will be in
+	 * the same order as this array, so you can iterate over it.
+	 * @param  {params.text} The message text to send
+	 * @param  {params.from} The message sender"s telephone number (or short code)
+	 * This must be a Catapult number that you own.
+	 * @param  {String} [params.to] Message recipient telephone number (or short code)
+	 * @param  {Array} [params.media] Json array containing list of media urls to be sent as content for an mms.
+	 * Valid URLs are: https://api.catapult.inetwork.com/v1/users/<user-id>/media/
+	 * We also support media URLs that are external to Bandwidth API, http:// or https:// format:
+	 * Example: http://customer-web-site.com/file.jpg
+	 * @param  {String} [params.callbackUrl] The complete URL where the events related to the
+	 * outgoing message will be sent
+	 * @param  {Number} [params.callbackTimeout] Determine how long should the platform wait for
+	 * callbackUrl"s response before timing out (milliseconds)
+	 * @param  {String} [params.fallbackUrl] The server URL used to send message events
+	 * if the request to callbackUrl fails
+	 * @param  {String} [params.tag] A string that will be included in the callback events of the message
+	 * @param  {String} [params.receiptRequested=none] Requested receipt option for outbound messages:
+	 * `none` `all` `error`
+	 * @param  {Function} [callback] A callback for the array of ExtendedMessageResponse
+	 * @returns [ExtendedMessageResponse] A promise for the array of ExtendedMessageResponse
```

Comment:
Should be `{ExtendedMessageResponse}` the array type is handled in the `@class` definition below... ideally we'd want to use `@typedef` but the `@typedef` doesn't populate the global namespace by default with our jsdoc tooling, so there is not clear way to access that return type w/o clicking around.